### PR TITLE
Fix golint issues in pkg/controller/statefulset

### PR DESCRIPTION
**Pull request by ScrapCodes**
**What type of PR is this?**

/kind cleanup
/kind api-change

**What this PR does / why we need it**:
Cleans up all the golint issues in `pkg/controller/statefulset`

**Which issue(s) this PR fixes**:
#68026

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

Currently, I am not sure, what goes in here. This patch does change some exposed functions.

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
thanks for the PR.

/ok-to-test
/priority backlog

**Comment by ScrapCodes** [comment]
Hi @oomichi Thank you !, are there any next steps for me?

**Comment by oomichi** [comment]
@ScrapCodes 
As https://github.com/kubernetes/kubernetes/pull/94082#issuecomment-675481724 you need to get approvals from approvers of
```
    hack/OWNERS
    pkg/controller/statefulset/OWNERS
    test/OWNERS
```

**Comment by ScrapCodes** [comment]
@dims would you like to take a look ?

**Comment by ScrapCodes** [comment]
/retest all

Trying to rerun tests, as some time has elapsed, since all the tests above passed.

**Comment by ScrapCodes** [comment]
Hi @liggitt, you are one of the owners listed. 

Can you please take a look at this pr? In case, this finds you busy then can you suggest/ defer it to someone who might want to take a look.

Thanks !!

File: pkg/controller/statefulset/stateful_set.go
Comment (soltysh) [inline pkg/controller/statefulset/stateful_set.go:52]: Please don't change those, it's not desirable to blindly follow golint failures.

**Review by soltysh — CHANGES_REQUESTED** [review]
/hold

File: pkg/controller/statefulset/stateful_set.go
Comment (ScrapCodes) [reply pkg/controller/statefulset/stateful_set.go:52]: Thank you so much for taking a look, I was myself in a similar impression. And asked the same question (here)[https://github.com/kubernetes/kubernetes/issues/68026#issuecomment-675431654].

 Is there is a way to ignore only a specific failure of Go lint, I am aware that we can have a comment to ignore an entire file. 

Shall I just close this PR? or is there a reason to hold - just trying to be sure.

Comment (soltysh) [reply pkg/controller/statefulset/stateful_set.go:52]: If this is the only change in your PR, then I guess so.

**Pull request closed without merging**
